### ISSUE TITLE
how to convert yolov4-csp,mish into .pth format?

### ISSUE THREAD

**Issue by akashAD98** [Tianxiaomo/pytorch-YOLOv4#424]
can you please make a code for these models? thanks in advance

**Comment by frenky-strasak** [comment]
@akashAD98 Did you find any solution?

**Comment by akashAD98** [comment]
yes i did , first clone the original GitHub repo & use this code

```
from tool import darknet2pytorch
import torch
import os
from torch.autograd import Variable
import torch.onnx
import torchvision
import torch
model_param_path = r"converted_models"
# load weights from darknet format
model_config = os.path.join(model_param_path,'yolov4x-mish_custom.cfg')
model = darknet2pytorch.Darknet(model_config, inference=True)
weights = os.path.join(model_param_path,"yolov4x-mish_custom.weights")
model.load_weights(weights)

# save weights to pytorch format
torch.save(model.state_dict(), os.path.join(model_param_path,"savedmodel.pth"))

# reload weights from pytorch format
model_pt = darknet2pytorch.Darknet(model_config, inference=True)
model_pt.load_state_dict(torch.load(os.path.join(model_param_path,"savedmodel.pth")))
dummy_input = Variable(torch.randn(10, 3, 416, 416))     #10, 3, 416, 416))
torch.onnx.export(model, dummy_input, "apperals_chaneel.onnx",input_names = ['input'], output_names = ['output',"boxes"],
 dynamic_axes= {'input' : {0 : "batch"},'output' : {0 : "batch"},'boxes' : {0 : "boxes"}})
```

**Comment by frenky-strasak** [comment]
@akashAD98 thank you for your response!

My goal is to create onnx model from yolov4-csp and yolov4-mish.
I used your code to generate the onnx model "apperals_chaneel.onnx". But the result is the same like from [this command](https://github.com/Tianxiaomo/pytorch-YOLOv4#3-darknet2onnx):
`python demo_darknet2onnx.py <cfgFile> <weightFile> <imageFile> <batchSize>` (command from this repository):
![predictions_onnx](https://user-images.githubusercontent.com/3219633/124780245-bdcb8780-df42-11eb-9500-c46a2be20e6a.jpg)
There are many wrong bboxes.

Also this implementation does not contain "logistic" function [here](https://github.com/Tianxiaomo/pytorch-YOLOv4/blob/4ccef0ec8fe984e059378813e33b3740929e0c19/tool/darknet2pytorch.py#L267) required for yolov4-csp, so I tried to implement there, but the wrong bboxes are still there.

Do you have any idea what's wrong here?
Thank you!

**Comment by akashAD98** [comment]
@frenky-strasak yes there is issue ,i was able to convert it into .pth but for .onnx im also  getting the same issue, here we need to write our own code for mish function, i havnt did it, i attached a few issues here, if you get something please let me know.
[https://github.com/onnx/onnx/issues/3475](url)
[https://github.com/hunglc007/tensorflow-yolov4-tflite/issues/280](url)
[https://github.com/onnx/onnx/issues/2818](url)

**Comment by frenky-strasak** [comment]
@akashAD98 , I found this [repo](https://github.com/linghu8812/tensorrt_inference/tree/master/Yolov4), where the converting yolov4-csp darknet to onnx format works.

**Comment by akashAD98** [comment]
Thanks.
 have you tried converting yolov4 CSP into .PB format ? I'm looking for
this script.

**Comment by akashAD98** [comment]
@frenky-strasak i was able to convert yolox-mish model into onnx, Do you have the script to testing inferencing of onxx models??
Thanks
